types.go

```diff
@@ -521,3 +519,117 @@ func ParseAutocompletePlaceType(placeType string) (AutocompletePlaceType, error)
+// https://cloud.google.com/maps-platform/user-guide/product-changes/#places
+type PlaceDetailsFieldMask string
+
+// The individual Place Details Field Masks.
+const (
+	PlaceDetailsFieldMaskPlaceID                  = PlaceDetailsFieldMask("place_id")
+	PlaceDetailsFieldMaskName                     = PlaceDetailsFieldMask("name")
+	PlaceDetailsFieldMaskType                     = PlaceDetailsFieldMask("type")
+	PlaceDetailsFieldMaskAddressComponents        = PlaceDetailsFieldMask("address_components")
+	PlaceDetailsFieldMaskFormattedAddress         = PlaceDetailsFieldMask("formatted_address")
+	PlaceDetailsFieldMaskURL                      = PlaceDetailsFieldMask("url")
+	PlaceDetailsFieldMaskUTCOffset                = PlaceDetailsFieldMask("utc_offset")
+	PlaceDetailsFieldMaskPermanentlyClosed        = PlaceDetailsFieldMask("permanently_closed")
+	PlaceDetailsFieldMaskGeometry                 = PlaceDetailsFieldMask("geometry")
+	PlaceDetailsFieldMaskGeometryLocation         = PlaceDetailsFieldMask("geometry.location")
+	PlaceDetailsFieldMaskGeometryViewport         = PlaceDetailsFieldMask("geometry.viewport")
+	PlaceDetailsFieldMaskPhoto                    = PlaceDetailsFieldMask("photo")
+	PlaceDetailsFieldMaskPhotoReference           = PlaceDetailsFieldMask("photo.photo_reference")
+	PlaceDetailsFieldMaskPhotos                   = PlaceDetailsFieldMask("photos")
+	PlaceDetailsFieldMaskIcon                     = PlaceDetailsFieldMask("icon")
+	PlaceDetailsFieldMaskTypes                    = PlaceDetailsFieldMask("types")
+	PlaceDetailsFieldMaskADRAddress               = PlaceDetailsFieldMask("adr_address")
+	PlaceDetailsFieldMaskScope                    = PlaceDetailsFieldMask("scope")
+	PlaceDetailsFieldMaskVicinity                 = PlaceDetailsFieldMask("vicinity")
+	PlaceDetailsFieldMaskOpeningHours             = PlaceDetailsFieldMask("opening_hours")
+	PlaceDetailsFieldMaskOpeningHoursWeekdayText  = PlaceDetailsFieldMask("opening_hours.weekday_text")
+	PlaceDetailsFieldMaskOpeningHoursOpenNow      = PlaceDetailsFieldMask("opening_hours.open_now")
+	PlaceDetailsFieldMaskOpeningHoursPeriod       = PlaceDetailsFieldMask("opening_hours.period")
+	PlaceDetailsFieldMaskWebsite                  = PlaceDetailsFieldMask("website")
+	PlaceDetailsFieldMaskFormattedPhoneNumber     = PlaceDetailsFieldMask("formatted_phone_number")
+	PlaceDetailsFieldMaskInternationalPhoneNumber = PlaceDetailsFieldMask("international_phone_number")
+	PlaceDetailsFieldMaskPriceLevel               = PlaceDetailsFieldMask("price_level")
+	PlaceDetailsFieldMaskRatings                  = PlaceDetailsFieldMask("rating")
+	PlaceDetailsFieldMaskReviews                  = PlaceDetailsFieldMask("reviews")
+)
+
+// ParsePlaceDetailsFieldMask will parse a string representation of
+// PlaceDetailsFieldMask.
+func ParsePlaceDetailsFieldMask(placeDetailsFieldMask string) (PlaceDetailsFieldMask, error) {
+	switch strings.ToLower(placeDetailsFieldMask) {
```

Comment:
You could make this mildly less annoying by removing the individual returns, because you know the value is "valid".

aka:

```go
src := strings.ToLower(placeDetailsFieldMask)
switch src {
  case "place_id":
  case "name":
  case "type":
  // etc
  default:
     return PlaceDetailsFieldMask(""), errors.New("...");
}
return PlaceDetailsFieldMask(src), nil
```

But up to you

Reply:
Given the way things are shifting, I'm appreciative of the redundancy.